src/test/mir-opt/if_condition_int.dont_remove_comparison.SimplifyComparisonIntegral.diff

```diff
@@ -18,16 +18,12 @@
           StorageLive(_2);                 // scope 0 at $DIR/if-condition-int.rs:44:9: 44:10
           StorageLive(_3);                 // scope 0 at $DIR/if-condition-int.rs:44:13: 44:14
           _3 = _1;                         // scope 0 at $DIR/if-condition-int.rs:44:13: 44:14
--         _2 = Eq(move _3, const 17_i8);   // scope 0 at $DIR/if-condition-int.rs:44:13: 44:20
--         StorageDead(_3);                 // scope 0 at $DIR/if-condition-int.rs:44:19: 44:20
--         switchInt(_2) -> [false: bb2, otherwise: bb1]; // scope 1 at $DIR/if-condition-int.rs:46:9: 46:14
-+         _2 = Eq(_3, const 17_i8);        // scope 0 at $DIR/if-condition-int.rs:44:13: 44:20
-+         nop;                             // scope 0 at $DIR/if-condition-int.rs:44:19: 44:20
-+         switchInt(move _3) -> [17_i8: bb1, otherwise: bb2]; // scope 1 at $DIR/if-condition-int.rs:46:9: 46:14
+          _2 = Eq(move _3, const 17_i8);   // scope 0 at $DIR/if-condition-int.rs:44:13: 44:20
+          StorageDead(_3);                 // scope 0 at $DIR/if-condition-int.rs:44:19: 44:20
+          switchInt(_2) -> [false: bb2, otherwise: bb1]; // scope 1 at $DIR/if-condition-int.rs:46:9: 46:14
```

Comment:
This one is interesting... this change by itself enabled an optimization to trigger more often